app/scripts/controllers/createFromURL.js

```diff
@@ -0,0 +1,191 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:CreateFromURLController
+ * @description
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('CreateFromURLController', function ($scope, $routeParams, $location, $filter, AuthService, DataService, AlertMessageService, Navigate, ProjectsService ) {
+    AuthService.withUser();
+
+    AlertMessageService.getAlerts().forEach(function(alert) {
+      $scope.alerts[alert.name] = alert.data;
+    });
+    AlertMessageService.clearAlerts();
+
+    $scope.alerts = {};
+    $scope.selected = {};
+
+    var alertInaccessibleNamespace = function(namespace, e) {
+      $scope.alerts.inaccessibleNamespace = {
+        type: "error",
+        message: "You do not have access to the namespace \"" + namespace + "\".",
+        details: "Reason: " + $filter('getErrorDetails')(e)
+      };
+    };
+
+    var alertInvalidImageStream = function(imageStream, e) {
+      $scope.alerts.invalidImageStream = {
+        type: "error",
+        message: "The requested image stream \"" + imageStream + "\" is not valid.",
+        details: "Reason: " + $filter('getErrorDetails')(e)
+      };
+    };
+
+    var alertInvalidImageTag = function(imageTag, e) {
+      $scope.alerts.invalidImageTag = {
+        type: "error",
+        message: "The requested image stream tag \"" + imageTag + "\" is not valid.",
+        details: "Reason: " + $filter('getErrorDetails')(e)
+      };
+    };
+
+    var alertInvalidNamespace = function(namespace) {
+      $scope.alerts.invalidNamespace = {
+        type: "error",
+        message: namespace ?
+                  "Resources from the namespace \"" + namespace + "\" are not permitted." :
+                  "A valid namespace is required."
```

Comment:
Let's open a follow-on issue to add a doc link here for whitelisting namespaces.